scripts/spack_packages/umpire/package.py

```diff
@@ -112,6 +113,8 @@ class Umpire(CMakePackage, CudaPackage):
     conflicts('+openmp', when='+hip')
     conflicts('+openmp_target', when='+hip')
     conflicts('+deviceconst', when='~hip~cuda')
+    conflicts('~mpi', when='+posix_shmem', msg='Shared Memory Allocator requires MPI')
```

Comment:
This is saying posix_shmem requires MPI, but this is not enfornced at CMake time.

Reply:
@davidbeckingsale, I have attempted to addres this with https://github.com/LLNL/Umpire/pull/553/commits/650df6707640250f5ee9dce873dbf3d650194a5b

Please let me know if you think that is sufficient.  It seems to work.  The only thing I don't like is that it does not really offer any feedback to the user other than "Shared Memory is Disabled" if the user does not also specify ENABLE_MPI.